src/acceptance-test/java/de/zalando/aruha/nakadi/webservice/CompressedEventPublishingAT.java

```diff
@@ -0,0 +1,62 @@
+package de.zalando.aruha.nakadi.webservice;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import de.zalando.aruha.nakadi.config.JsonConfig;
+import de.zalando.aruha.nakadi.domain.EventType;
+import org.apache.http.HttpStatus;
+import org.json.JSONObject;
+import org.junit.Test;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.zip.GZIPOutputStream;
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.http.ContentType.JSON;
+import static de.zalando.aruha.nakadi.utils.TestUtils.buildEventType;
+import static de.zalando.aruha.nakadi.utils.TestUtils.randomValidEventTypeName;
+import static java.text.MessageFormat.format;
+import static javax.ws.rs.core.HttpHeaders.CONTENT_ENCODING;
+
+public class CompressedEventPublishingAT extends BaseAT {
```

Comment:
Very good point, as it was actually returning 500 in that case :-D (which is not what we want)
I now fixed that and added AT for that.